packages/apps/graph/src/services/chainweb-node/mempool.ts

```diff
@@ -0,0 +1,100 @@
+import type { Signer } from '@prisma/client';
+import { chainIds } from '@utils/chains';
+import { dotenv } from '@utils/dotenv';
+
+export class MempoolError extends Error {
+  public mempoolError: any;
+
+  constructor(message: string, mempoolError?: any) {
+    super(message);
+    this.mempoolError = mempoolError;
+  }
+}
+
+export async function mempoolGetPending() {
+  try {
+    process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
```

Comment:
I mentioned this in the other PR as well but is there no other way around this? This feels like a bad idea

Reply:
I think I would need to have a certificate if we don't want to disable TSL/SSL certificate verification. @enobayram do you maybe know if this would be possible to obtain? Without it I can't make requests to the mempool

Reply:
As far as I know, the node's service port doesn't serve the mempool endpoint, but I'm not sure why, (might be worth asking the chainweb team to expose it there).

As long as we need to get that information from the p2p port, and if the node isn't configured with a domain name and a valid certificate for that domain, you need to accept an unauthorized certificate. That said, if I'm not mistaken, this code snippet globally disables the TLS authority check in a rather ad-hoc manner, on which I agree with @MRVDH that it seems very shady.

My 2 cents here would be to make a check at `graph` start up whether the configured p2p endpoint is reachable. If the `graph` operator wants to make `graph` talk securely with their `chainweb-node`, then they can set up a proper domain and a signed SSL certificate for it, then `graph` can just talk to it without messing with TLS settings. Or maybe the `graph` operator is running the node and `graph` in a secure network, in which case they don't care about a signed SSL certificate. In this case, your reachability check would fail at `graph` start up and at that time you can inform the user about the situation and tell them that they should either configure their node with a domain or explicitly run `graph` with `NODE_TLS_REJECT_UNAUTHORIZED=0` in the environment.